.github/workflows/rspec.yml

```
@@ -1,5 +1,11 @@
 name: RSpec
-on: [push]
+on: 
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - '*'
```

This would trigger the workflow on any push to any branch, which you probably don't want. Better to only trigger the action on push to `main` & `develop`.
```suggestion
on:
  pull_request:
  push:
    branches:
      - main
      - develop
```